de.metas.contracts/src/main/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportHelper.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * 
+ */
+package de.metas.contracts.flatrate.impexp;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Properties;
+
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.bpartner.service.IBPartnerDAO;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.model.PlainContextAware;
+import org.adempiere.util.Services;
+import org.adempiere.util.time.SystemTime;
+import org.compiere.model.I_AD_User;
+import org.compiere.model.I_C_UOM;
+import org.compiere.model.I_M_Product;
+import org.slf4j.Logger;
+
+import com.google.common.collect.Ordering;
+
+import de.metas.adempiere.model.I_C_BPartner_Location;
+import de.metas.contracts.IFlatrateBL;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.contracts.model.I_I_Flatrate_Term;
+import de.metas.contracts.model.X_C_Flatrate_Term;
+import de.metas.logging.LogManager;
+import de.metas.product.IProductBL;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * @author cg
+ *
+ */
+/* package */ class FlatrateTermImportHelper
+{
+	public static FlatrateTermImportHelper newInstance()
```

Comment:
may be is makes sense to pass `FlatrateTermImportProcess` as parameter and kick out the `setProcess` method